File: 2021/2-1. Dive/src/main.cpp
#include <fstream>
#include <iostream>
#include <string>

int main()
{
    std::ifstream infile("input.txt");

    if (!infile.is_open())
    {
        std::cerr << "ERROR! Could not open input.txt!\n";
        return -1;
    }

    int x = 0;
    int depth = 0;

    std::string type;
    int ammount;

    while (infile >> type >> ammount)
    {
        if (type == "forward")
            x += ammount;
        else if (type == "up")
            depth -= ammount;
        else
            depth += ammount;
    }

    std::cout << "Answer: " << x * depth << '\n';

    return 0;
}

File: 2021/2-1. Dive/Python/main.py
infile = open("input.txt")

pos = 0
depth = 0

for line in infile:
    spacePos = line.find(' ')

    command = line[:spacePos]
    ammount = int(line[spacePos:])

    if command == "forward":
        pos += ammount
    elif command == "up":
        depth -= ammount
    else:
        depth += ammount

print("Answer:", pos * depth)

infile.close()